### PR TITLE
Apps with Prometheus enabled needs access to istio-system default SA

### PR DESCRIPTION
**Pull request by linemos**
Solves #81

### CODE REVIEW

File: pkg/resourcecreator/istio.go [hunk 1]
```diff
@@ -68,6 +64,17 @@ func ServiceRoleBinding(app *nais.Application) (*istio_crd.ServiceRoleBinding, e
 		})
 	}
 
+	if app.Spec.Prometheus.Enabled {
+		rules = append(rules, nais.AccessPolicyGressRule{
+			Namespace:   "istio-system",
+			Application: "default",
```
Comment (ambientsound) [inline pkg/resourcecreator/istio.go]: Curious why the application is named "default"?

**Comment by ambientsound** [comment]
Suggest using `const` values here, as well.

File: pkg/resourcecreator/istio.go (hunk 1 above)
Comment (linemos) [reply pkg/resourcecreator/istio.go]: Bare for å opprette binding til `cluster.local/ns/istio-system/sa/default`.

Comment (terjesannum) [reply pkg/resourcecreator/istio.go]: Dunno why it's called default, but that is what's used in the istio prometheus installation:
https://github.com/istio/istio/blob/d8b9a10661c1d1ded591cbef39303c83fb1ed56f/install/kubernetes/helm/istio/charts/prometheus/templates/deployment.yaml#L70

**Comment by jhrv** [comment]
This will effectively grant prometheus full access to the entire application surface, and will make prometheus a high value target to compromise. Since we know what endpoint and http verb prometheus will be calling the application with, maybe we should create a more explicit role?

**Pull request merged**